public/app/features/storage/StoragePage.tsx

```diff
@@ -0,0 +1,204 @@
+import { css } from '@emotion/css';
+import React, { useMemo } from 'react';
+import { useAsync } from 'react-use';
+
+import { DataFrame, GrafanaTheme2, isDataFrame, ValueLinkConfig } from '@grafana/data';
+import { locationService } from '@grafana/runtime';
+import { useStyles2, IconName, Spinner, TabsBar, Tab, Button, HorizontalGroup } from '@grafana/ui';
+import Page from 'app/core/components/Page/Page';
+import { useNavModel } from 'app/core/hooks/useNavModel';
+import { GrafanaRouteComponentProps } from 'app/core/navigation/types';
+
+import { AddRootView } from './AddRootView';
+import { Breadcrumb } from './Breadcrumb';
+import { ExportView } from './ExportView';
+import { FileView } from './FileView';
+import { FolderView } from './FolderView';
+import { RootView } from './RootView';
+import { getGrafanaStorage } from './helper';
+import { StorageView } from './types';
+
+interface RouteParams {
+  path: string;
+}
+
+interface QueryParams {
+  view: StorageView;
+}
+
+interface Props extends GrafanaRouteComponentProps<RouteParams, QueryParams> {}
+
+export default function StoragePage(props: Props) {
+  const styles = useStyles2(getStyles);
+  const navModel = useNavModel('storage');
+  const path = props.match.params.path ?? '';
+  const view = props.queryParams.view ?? StorageView.Data;
+  const setPath = (p: string, view?: StorageView) => {
+    let url = ('/admin/storage/' + p).replace('//', '/');
+    if (view && view !== StorageView.Data) {
+      url += '?view=' + view;
+    }
+    locationService.push(url);
+  };
+
+  const listing = useAsync((): Promise<DataFrame | undefined> => {
+    return getGrafanaStorage()
+      .list(path)
+      .then((frame) => {
+        if (frame) {
+          const name = frame.fields[0];
+          frame.fields[0] = {
+            ...name,
+            getLinks: (cfg: ValueLinkConfig) => {
+              const p = path + '/' + name.values.get(cfg.valueRowIndex ?? 0);
+              return [
+                {
+                  title: 'Open XYZ',
```

Comment:
```suggestion
                  title: `Open ${getFilename(p)}`,
```

and something like 


```
const getFilename = (path: string): string => {
  const lastSlashIdx = path.lastIndexOf('/');
  if (lastSlashIdx <= 0) {
    return path;
  }

  return path.substring(lastSlashIdx + 1);
};
```
?